cli/flags.rs

```diff
@@ -2592,7 +2595,7 @@ fn permission_args_parse(flags: &mut Flags, matches: &clap::ArgMatches) {
     flags.allow_ffi = Some(vec![]);
     flags.allow_hrtime = true;
   }
-  if matches.is_present("no-prompt") {
+  if env::var("DENO_NO_PROMPT").is_ok() || matches.is_present("no-prompt") {
```

Comment:
I think this reading of env variable should be done in `ProcState::build`; otherwise the tests in `flags.rs` will depend on the current environment.

Reply:
this imo should be rather written with https://docs.rs/clap/latest/clap/struct.Arg.html#method.env, though @bartlomieju does have a point...

Reply:
I like @crowlKats's solution since that's idiomatic. Is there a way to configure the env for `cargo test` where it can be unset?

Reply:
See https://github.com/denoland/deno/pull/12769#discussion_r749553155